src/v/cloud_storage/topic_manifest.cc

```diff
@@ -12,21 +12,193 @@
+                if (boost::conversion::try_lexical_convert(
+                      sv, compaction_strategy_res)) {
+                    _properties.compaction_strategy = compaction_strategy_res;
+                }
+            } else if ("timestamp_type" == _key) {
+                model::timestamp_type timestamp_type_res;
+                if (boost::conversion::try_lexical_convert(
+                      sv, timestamp_type_res)) {
+                    _properties.timestamp_type = timestamp_type_res;
+                }
+            } else {
+                return false;
+            }
+            _state = expectKey;
+            return true;
+        case expectManifestStart:
+        case expectKey:
+            return false;
+        }
+    }
+
+    bool Uint(unsigned u) { return Uint64(u); }
+
+    bool Uint64(uint64_t u) {
+        switch (_state) {
+        case expectValue:
+            if ("version" == _key) {
+                _version = static_cast<topic_manifest_version>(u);
+                if (_version != topic_manifest_version::v1) {
+                    throw std::runtime_error(
+                      "topic manifest version not supported");
+                }
+            } else if ("partition_count" == _key) {
+                _partition_count = u;
+            } else if ("replication_factor" == _key) {
+                _replication_factor = u;
+            } else if ("revision_id" == _key) {
+                _revision_id = model::initial_revision_id(u);
+            } else if ("segment_size" == _key) {
+                _properties.segment_size = u;
+            } else if ("retention_bytes" == _key) {
+                _properties.retention_bytes = tristate<size_t>(u);
+            } else if ("retention_duration" == _key) {
+                _properties.retention_duration
+                  = tristate<std::chrono::milliseconds>(
+                    std::chrono::milliseconds(u));
+            } else {
+                return false;
+            }
+            _state = expectKey;
+            return true;
+        case expectManifestStart:
+        case expectKey:
+            return false;
+        }
+    }
+
+    bool EndObject(rapidjson::SizeType /*size*/) {
+        if (_state == expectKey && all_required_complete()) {
+            _topic_config = cluster::topic_configuration(
+              _namespace.value(),
+              _topic.value(),
+              _partition_count.value(),
+              _replication_factor.value());
+            _topic_config->properties = _properties;
+            return true;
+        }
+        return false;
+    }
+
+    bool Null() {
+        if (_state == expectValue) {
+            _state = expectKey;
+            return true;
+        }
+        return false;
+    }
+
+    bool Default() { return false; }
+
+    bool all_required_complete() {
+        return _version && _namespace && _topic && _partition_count
+               && _replication_factor && _revision_id;
+    }
+
+    enum State {
```

Comment:
can we use `enum class` here for strong type safety?

Reply:
Fixed